lib/xdrgen/generators/go.rb

```diff
@@ -336,73 +332,74 @@ def render_union(out, union)
         out.break
       end
 
-      def render_struct_encode_to_interface(out, struct)
+      def render_struct_marshal_xdr_interface(out, struct)
         name = name(struct)
-        out.puts "// EncodeTo encodes this value using the Encoder."
-        out.puts "func (s *#{name}) EncodeTo(e *xdr.Encoder) error {"
-        out.puts "  var err error"
+        out.puts "// MarshalXDR implements xdr.Marshaler."
+        out.puts "func (s #{name}) MarshalXDR(e *xdr.Encoder) (int, error) {"
```

Comment:
Unfortunately, I am forced to call by value (note the lack of `*`) when switching to `MarshallXDR)()` since, otherwise, `xdr.Encoder()` cannot invoke the  method on unadresable values.

This implies calling by value, which can (and as I observed, it does) cause allocations, betraying the purpose of doing all this (which is increasing performance).